### ISSUE TITLE
🐛 Invalid document: Expected end tag of kind Embedded but found Group.

### ISSUE THREAD

**Issue by wojtekmaj** [biomejs/biome#9112]
### Environment information

<details>

```bash
CLI:
  Version:                      2.4.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.11.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         yarn/4.10.3

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    unset

Workspace:
  Open Documents:               0
```

</details>

### What happened?

https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIABzAHQAeQBsAGUAZAAgAGYAcgBvAG0AIAAnAHMAdAB5AGwAZQBkAC0AYwBvAG0AcABvAG4AZQBuAHQAcwAnADsACgAKAGUAeABwAG8AcgB0ACAAYwBvAG4AcwB0ACAAVwByAGEAcABwAGUAcgAgAD0AIABzAHQAeQBsAGUAZAAuAGQAaQB2ADwAewAKACAAIAB2AGEAcgBpAGEAbgB0ADoACgAgACAAIAAgAHwAIAAnAGYAbwBvACcACgAgACAAIAAgAHwAIAAoAHMAdAByAGkAbgBnACAAJgAgAFIAZQBjAG8AcgBkADwAbgBlAHYAZQByACwAIABuAGUAdgBlAHIAPgApADsACgB9AD4AYAAKACAAIAAgACAAaABlAGkAZwBoAHQAOgAgADIAMABwAHgAOwAKAGAAOwAKAA%3D%3D&ruleDomains.react=recommended

### Expected result

Biome to format CSS-in-JS when experimentalEmbeddedSnippetsEnabled is enabled

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by denyo** [comment]
I see the errors with different input. It looks like as soon as there is any jsx in the same file, it fails with
```
Error: Invalid document: Expected end tag of kind Embedded but found Group.
```
See: https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIABzAHQAeQBsAGUAZAAgAGYAcgBvAG0AIAAnAHMAdAB5AGwAZQBkAC0AYwBvAG0AcABvAG4AZQBuAHQAcwAnADsACgAKAC8ALwAgAGMAbwBtAG0AZQBuAHQAIABvAHIAIAByAGUAbQBvAHYAZQAgAEQAZQBtAG8AIABjAG8AbQBwAG8AbgBlAG4AdAAgAGEAbgBkACAAaQB0ACAAdwBvAHIAawBzAAoAZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABEAGUAbQBvACAAPQAgACgAKQAgAD0APgAgACgACgAJADwAQwBvAG4AdABhAGkAbgBlAHIAPgAKAAkACQA8AHMAcABhAG4AIAB7AC4ALgAuAHAAcgBvAHAAcwB9ACAALwA%2BAAoACQA8AC8AQwBvAG4AdABhAGkAbgBlAHIAPgAKACkACgAKAGMAbwBuAHMAdAAgAEMAbwBuAHQAYQBpAG4AZQByACAAPQAgAHMAdAB5AGwAZQBkAC4AZABpAHYAYAAKAAkAIAAgACAAIAAgAG0AaQBuAC0AaABlAGkAZwBoAHQAOgAgADQANAAwAHAAeAA7AAoAYAA%3D&ruleDomains.react=recommended

**Comment by JUSTIVE** [comment]
It appears that the same error is occurring in the latest version as well(Biome [1f30838]
https://github.com/biomejs/biome/tree/1f30838b288bf317bf88c9247da029d9a394d009)

**Comment by denyo** [comment]
It looks like both linked playgrounds from above are still failing with latest `2.4.10`.

**Comment by ematipico** [comment]
Closing. Both background works now.
